app/src/marketplace/containers/AvailableStreams/index.jsx

```diff
@@ -0,0 +1,58 @@
+// @flow
+
+import { useEffect } from 'react'
+import { connect } from 'react-redux'
+
+import type { StoreState } from '$shared/flowtype/store-state'
+import { getStreams as getStreamsAction } from '$mp/modules/streams/actions'
+import { selectStreams, selectFetchingStreams, selectStreamsError } from '$mp/modules/streams/selectors'
+import type { StreamList } from '$shared/flowtype/stream-types'
+import type { ErrorInUi } from '$shared/flowtype/common-types'
+
+export type StateProps = {
+    fetching: boolean,
+    error: ?ErrorInUi,
+    streams: ?StreamList,
+}
+
+export type DispatchProps = {
+    getStreams: () => void,
+}
+
+type Props = StateProps & DispatchProps & {
+    children: Function,
+}
+
+const AvailableStreams = ({
+    children,
+    fetching,
+    streams,
+    error,
+    getStreams,
+}: Props) => {
+    if (!children || typeof children !== 'function') {
+        throw new Error('children needs to be a function!')
+    }
+
+    useEffect(() => {
+        getStreams()
+    }, [getStreams])
+
+    return children ? children({
+        fetching,
+        error,
+        streams,
+    }) : null
+}
+
+export const mapStateToProps = (state: StoreState): StateProps => ({
```

Comment:
Should we use `react-redux` hooks instead? https://react-redux.js.org/next/api/hooks

Reply:
Yep, changed to use hooks in https://github.com/streamr-dev/streamr-platform/pull/601/commits/5de6bd03c1eb567d3a3d93f096026a349f79c4f1.